src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php

```diff
@@ -43,7 +43,8 @@ class Connection
     private $consumer;
     private $autoSetup;
     private $maxEntries;
-    private $couldHavePendingMessages = true;
+    private $claimTimeout = 600000;
+    private $couldHavePendingMessages = false;
```

Comment:
couldHavePendingMessages should be true because old messages of the current consumer should be read before continuing with new message this should be tested in testKeepGettingPendingMessages which was removed in your PR

Reply:
You are right. I prioritised pending messages from other consumers. I've changed the default value and readded the test.